iot-hub/Samples/service/FileUploadNotificationReceiverSample/FileUploadNotificationReceiverSample.cs

```diff
@@ -0,0 +1,128 @@
+            }
+        }
+
+        private async Task ReceiveFileUploadNotifications(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Listening for file upload notifications from the service.");
+
+            FileNotificationReceiver<FileNotification> notificationReceiver = _serviceClient.GetFileNotificationReceiver();
+
+            int totalNotificationsReceived = 0;
+            int totalNotificationsCompleted = 0;
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    FileNotification fileUploadNotification = await notificationReceiver.ReceiveAsync(s_notificationReceiverTimeout);
+
+                    if (fileUploadNotification == null)
+                    {
+                        _logger.LogInformation($"Did not receive any notification after {s_notificationReceiverTimeout.TotalSeconds} seconds.");
+                        continue;
+                    }
+
+                    totalNotificationsReceived++;
+
+                    _logger.LogInformation($"Received file upload notification.");
+                    _logger.LogInformation($"\tDeviceId: {fileUploadNotification.DeviceId ?? "N/A"}.");
+                    _logger.LogInformation($"\tFileName: {fileUploadNotification.BlobName ?? "N/A"}.");
+                    _logger.LogInformation($"\tEnqueueTimeUTC: {fileUploadNotification.EnqueuedTimeUtc}.");
+                    _logger.LogInformation($"\tBlobSizeInBytes: {fileUploadNotification.BlobSizeInBytes}.");
+
+                    _logger.LogInformation($"Marking notification for {fileUploadNotification.DeviceId} as complete.");
+
+                    // Mark the notification as completed.
+                    await notificationReceiver.CompleteAsync(fileUploadNotification);
+
+                    totalNotificationsCompleted++;
+
+                    _logger.LogInformation($"Successfully marked the notification for device {fileUploadNotification.DeviceId} as completed.");
+                }
+                catch (Exception e) when ((e is IotHubException) || (e is DeviceMessageLockLostException))
+                {
+                    _logger.LogWarning($"Caught a recoverable exception, will retry: {e.Message} - {e}");
+                }
+                catch (Exception)
+                {
+                    // An exception was caught that was not recoverable, the exception will be thrown to be handled by the caller of the method.
+                    throw;
+                }
+            }
+
+            _logger.LogInformation($"Total Notifications Received: {totalNotificationsReceived}.");
+            _logger.LogInformation($"Total Notifications Marked as Completed: {totalNotificationsCompleted}.");
+
+            _logger.LogInformation($"Closing the service client.");
+
+            await _serviceClient.CloseAsync();
```

Comment:
Would it make more sense to close the service client in a finally or possibly outside of this method? If someone picks this up and runs with it they may end up not closing the client here and getting exceptions later on.